web/src/js/components/FlowTable.tsx

```diff
@@ -60,15 +60,19 @@ export class PureFlowTable extends React.Component<
         this.onViewportUpdate();
 
         const selectedNewFlow =
-            this.props.selected && this.props.selected !== prevProps.selected;
+            this.props.selected[0] &&
+            this.props.selected[0] !== prevProps.selected[0];
         if (selectedNewFlow) {
             const { rowHeight, flows, selected } = this.props;
             const viewport = this.viewport.current!;
             const head = this.head.current;
 
             const headHeight = head ? head.offsetHeight : 0;
 
-            const rowTop = flows.indexOf(selected) * rowHeight + headHeight;
+            const selectedFlowIndex = flows.findIndex(
+                (flow) => flow.id === selected[0],
+            );
```

Comment:
Perf: The store maintains a flowId -> index mapping:
https://github.com/mitmproxy/mitmproxy/blob/6269898b434e1040c4ea46a425526c5d266c554b/web/src/js/ducks/utils/store.ts#L25

Reply:
Regarding this, I replaced that line with **listIndex**, and it worked fine. However, during testing, I encountered the following error:

"_Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops_"

I had to add this statement 
```
if (!shallowEqual(this.props.flowIndexes, prevProps.flowIndexes)) {
      this.onViewportUpdate();
}
```
which solved the issue but it introduces some inefficiency again. Do you have any idea?